### ISSUE TITLE
[gtk 3.20] broke geometry of window

### ISSUE THREAD

**Issue by realh** [realh/roxterm#126]
After upgrate gtk 3.0 to version 3.20 (gtk 3.20 + vte 0.44), roxterm does not accept no more geometry settings of the window - window will appear always the smallest as possible.

also 'roxterm --geometry 120x30' does nothing

Reported by: *anonymous

Original Ticket: [roxterm/bugs/125](https://sourceforge.net/p/roxterm/bugs/125)

**Comment by realh** [comment]
Although we've managed to get rid of the main symptom, it still isn't right, and now has the opposite problem in Wayland. I think replacing gtk_window_parse_geometry should be the next step. Also, gtk_widget_get_alloc*, and even trying to read the size from the GdkWindow, do not appear to be reliable. All the functions that need to calculate the size of the top-level required to accommodate a terminal of a desired size probably need to be done during "size-allocate" signal handlers, which seems successful so far (when toggling the menu and tab bars).

**Comment by realh** [comment]
I just can't fix this. We need to call gtk_window_set_default_size before showing it, but to call gtk_window_set_default_size we need to know the size of the "chrome" and the window decorations. But even though I've packed all the widgets and called gtk_window_show_all on them (except for the top-level window, but I've called gtk_widget_realize on it), reading the allocations still returns useless values for the window and the vbox. gtk_widget_get_preferred_width/height returns values that are a bit more realistic, but they're still wrong.
